src/main/rx/flysky.c

```diff
@@ -215,167 +198,126 @@ static bool isValidPacket (const uint8_t *packet) {
-        case FLYSKY_2A_PACKET_SETTINGS: // receiver settings
+    case FLYSKY_2A_PACKET_RC_DATA:
+    case FLYSKY_2A_PACKET_FS_SETTINGS: // failsafe settings
+    case FLYSKY_2A_PACKET_SETTINGS: // receiver settings
         if (isValidPacket(packet)) {
             checkRSSI();
             resetTimeout(timeStamp);
-
             const flySky2ARcDataPkt_t *rcPacket = (const flySky2ARcDataPkt_t*) packet;
-
             if (rcPacket->type == FLYSKY_2A_PACKET_RC_DATA) {
                 if (payload) {
-                    memcpy(payload, rcPacket->data, 2*FLYSKY_2A_CHANNEL_COUNT);
+                    memcpy(payload, rcPacket->data, 2 * FLYSKY_2A_CHANNEL_COUNT);
                 }
-
                 if (sendTelemetry) {
                     buildAndWriteTelemetry(packet);
                     sendTelemetry = false;
                     timeTxRequest = timeStamp;
                     waitTx = true;
                 }
-
                 result = RX_SPI_RECEIVED_DATA;
             }
-
             if (!waitTx) {
                 A7105WriteReg(A7105_0F_CHANNEL, getNextChannel(1));
             }
         }
         break;
-
-        case FLYSKY_2A_PACKET_BIND1:
-        case FLYSKY_2A_PACKET_BIND2:
+    case FLYSKY_2A_PACKET_BIND1:
+    case FLYSKY_2A_PACKET_BIND2:
         if (!bound) {
             resetTimeout(timeStamp);
-
             flySky2ABindPkt_t *bindPacket = (flySky2ABindPkt_t*) packet;
-
             if (bindPacket->rfChannelMap[0] != 0xFF) {
                 memcpy(rfChannelMap, bindPacket->rfChannelMap, FLYSKY_FREQUENCY_COUNT); // get TX channels
             }
-
             txId = bindPacket->txId;
             bindPacket->rxId = rxId;
             memset(bindPacket->rfChannelMap, 0xFF, 26); // erase channelMap and 10 bytes after it
-
             timeTxRequest = timeLastBind = timeStamp;
             waitTx = true;
-
             A7105WriteFIFO(packet, FLYSKY_2A_PAYLOAD_SIZE);
         }
         break;
-
-        default:
+    default:
         break;
     }
-
-    if (!waitTx){
+    if (!waitTx) {
         A7105Strobe(A7105_RX);
     }
     return result;
 }
 
-static rx_spi_received_e flySkyReadAndProcess (uint8_t *payload, const uint32_t timeStamp)
-{
+static rx_spi_received_e flySkyReadAndProcess (uint8_t *payload, const uint32_t timeStamp) {
     rx_spi_received_e result = RX_SPI_RECEIVED_NONE;
     uint8_t packet[FLYSKY_PAYLOAD_SIZE];
-
-    uint8_t bytesToRead = (bound) ? (5 + 2*FLYSKY_CHANNEL_COUNT) : (5);
+    uint8_t bytesToRead = (bound) ? (5 + 2 * FLYSKY_CHANNEL_COUNT) : (5);
     A7105ReadFIFO(packet, bytesToRead);
-
     const flySkyRcDataPkt_t *rcPacket = (const flySkyRcDataPkt_t*) packet;
-
     if (bound && rcPacket->type == FLYSKY_PACKET_RC_DATA && rcPacket->txId == txId) {
         checkRSSI();
         resetTimeout(timeStamp);
-
         if (payload) {
-            memcpy(payload, rcPacket->data, 2*FLYSKY_CHANNEL_COUNT);
+            memcpy(payload, rcPacket->data, 2 * FLYSKY_CHANNEL_COUNT);
```

Comment:
Codacy found an issue: [Does not check for buffer overflows when copying to destination (CWE-120).  Make sure destination can always hold the source data. ](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=6307959)